File: Examples/Ruby/Pages/get_page_in_specified_format.rb
require 'aspose_pdf_cloud'

class Page

  include AsposePDFCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @pdf_api = PdfApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  # Get page in specified format.
  def get_page_in_specified_format
    file_name = "Sample-Annotation.pdf"
    upload_file(file_name)

    page_number = 1
    format = "png"
    response = @pdf_api.get_page_with_format(file_name, page_number, format)
  end

end

page = Page.new()
puts page.get_page_in_specified_format

File: Examples/PHP/Pages/GetPageInSpecifiedFormat.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\PDF\PdfApi;
use Aspose\PDF\AsposeApp;

class Pages {

	public $pdfApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->pdfApi = new PdfApi();
	}

	public function getPageWithFormat() {
        // Upload file to Aspose Cloud Storage
		$fileName = "Sample-Annotation.pdf";
		Utils::uploadFile($fileName);

        $result = $this->pdfApi->GetPageWithFormat($fileName, $pageNumber=1, $format="png", $width = null, $height = null, $storage = "", $folder = "");        
        $fh = fopen(realpath(__DIR__ . '/../../..') . '/Data/Output/Test.gif', 'w');
        fwrite($fh, $result);
        fclose($fh);
	}
}

$pages = new Pages();
$pages->getPageWithFormat();

?>
//ExEnd:
